### PR TITLE
Add gcloud auth in user kubeconfig secret

### LINKED ISSUE #923: Add Updated GCP Auth Plugin in generated user kubeconfig
/kind feature

**Describe the solution you'd like**
I would like to add `gke-gcloud-auth-plugin` authentication in the generated user-kubeconfig secret. Currently when I try
to use that secret, It shows an error
```
error: The gcp auth plugin has been removed.
Please use the "gke-gcloud-auth-plugin" kubectl/client-go credential plugin instead.
See https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke for further details
```

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by Shaad7**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

This PR adds `gcloud-auth` authentication in the user kubeconfig secret.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #923 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:
```
Add `gcloud-auth` authentication
```

### CODE REVIEW

File: cloud/services/container/clusters/kubeconfig.go [hunk 1]
```diff
@@ -108,18 +108,15 @@ func (s *Service) createUserKubeconfigSecret(ctx context.Context, cluster *conta
...
-		Name: "gcp",
-		Config: map[string]string{
-			"cmd-args":   "config config-helper --format=json",
-			"cmd-path":   "gcloud",
-			"expiry-key": "'{.credential.token_expiry}'",
-			"token-key":  "'{.credential.access_token}'",
-		},
+	execConfig := &api.ExecConfig{
+		APIVersion:         "client.authentication.k8s.io/v1beta1",
+		Command:            "gke-gcloud-auth-plugin",
```
Comment (richardcase) [inline cloud/services/container/clusters/kubeconfig.go:113]: @Shaad7 - should we only be using this from v1.26?

Comment (Shaad7) [reply cloud/services/container/clusters/kubeconfig.go:113]: Yes. The auth changed in GKE version 1.26

Comment (Shaad7) [reply cloud/services/container/clusters/kubeconfig.go:113]: @richardcase Should I add a condition to check the GKE version and generate kubeconfig according to the version ?

Comment (richardcase) [reply cloud/services/container/clusters/kubeconfig.go:113]: If it's version specific, then I think we will need to. wdyt?

Comment (Shaad7) [reply cloud/services/container/clusters/kubeconfig.go:113]: It is not version specific. The update works on all the versions and the old one does not. So, I think we are good to go with the PR.

**Comment by richardcase** [comment]
From my side:
/approve

For lgtm:

/assign cpanato

File: cloud/services/container/clusters/kubeconfig.go (hunk 1 above)
Comment (richardcase) [reply cloud/services/container/clusters/kubeconfig.go:113]: Thanks for the clarification, and sorry its taken so long to get back to you.

**Review by cpanato — APPROVED** [review]
/lgtm

thanks

**Pull request merged**